File: 377-combination-sum-iv/377-combination-sum-iv.cpp
class Solution {
public:
    int combinationSum4(vector<int>& nums, int t) {
        vector<unsigned int> dp(t + 1, 0);
        sort(nums.begin(), nums.end());
        dp[0] = 1;
        for(int i = 0; i < t; i++) {
            for(int e: nums) {
                if(i + e <= t) dp[i + e] += dp[i];
                else break;
            }
        }
        return dp[t];
    }
};

File: 377-combination-sum-iv/377-combination-sum-iv.rs
impl Solution {
    pub fn combination_sum4(nums: Vec<i32>, t: i32) -> i32 {
        let mut dp = vec![0 as u32; (t + 1) as usize];
        dp[0] = 1;
        for i in 1..(t + 1) {
            for e in &nums {
                if(i - e >= 0) {
                    dp[i as usize] += dp[(i - e) as usize];
                }
            }
        }
        return dp[t as usize] as i32;
    }
}